agents/agents-ext/src/jvmMain/kotlin/ai/koog/agents/ext/tool/shell/JvmShellCommandExecutor.kt

```diff
@@ -0,0 +1,109 @@
+package ai.koog.agents.ext.tool.shell
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.coroutines.withTimeout
+import java.io.File
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Shell command executor using ProcessBuilder for JVM platforms.
+ *
+ * @see ShellCommandExecutor
+ */
+public class JvmShellCommandExecutor : ShellCommandExecutor() {
+
+    private companion object {
+        val IS_WINDOWS = System.getProperty("os.name")
+            .lowercase()
+            .contains("win")
+    }
+
+    /**
+     * Executes a shell command and returns combined output and exit code.
+     *
+     * @param command Shell command string to execute
+     * @param workingDirectory Working directory, or null to use current directory
+     * @param timeoutSeconds Maximum execution time in seconds, or null for no timeout
+     * @return [ExecutionResult] containing combined stdout/stderr output and process exit code
+     */
+    override suspend fun execute(
+        command: String,
+        workingDirectory: String?,
+        timeoutSeconds: Int?
+    ): ExecutionResult = withContext(Dispatchers.IO) {
+        val shellCommand = if (IS_WINDOWS) {
+            listOf("cmd.exe", "/c", command)
+        } else {
+            listOf("bash", "-c", command)
```

Comment:
I would update here for "/bin/bash -c" as well.

Do you need to keep compatibility between different shells? Alternative more universal approach would be using "/bin/sh -c" here instead.